pg_shard.c

```diff
@@ -1107,6 +1105,17 @@ PgShardExecutorStart(QueryDesc *queryDesc, int eflags)
 			bool topLevel = true;
 			LOCKMODE lockMode = NoLock;
 			EState *executorState = NULL;
+			bool zeroShardQuery = list_length(distributedPlan->taskList) == 0;
+
+			/* if query involves zero shards, just let it hit local table */
+			if (zeroShardQuery)
+			{
+				Plan *originalPlan = distributedPlan->originalPlan;
+				plannedStatement->planTree = originalPlan;
+
+				PgShardExecutorStart(queryDesc, eflags);
```

Comment:
Hmm. Instead of PgShardExecutorStart calling itself (and then returning), can't we have a check and then fall to calling the hooks explicitly?
We already have a selectFromMultipleShards statement, so I was thinking of something like:
```
if (zeroShardQuery)
{
   // Comment describing why we do this, and why it works.
  // replace distributed plan with original plan 
  if (hook) ... (call hook)
   else standard_ExecutorStart();
}
else if (multiShardQuery)
{

}
else // singleShardQuery
{
 
}
```

This is then more readable, and I don't have to figure out after-effects of calling the function again.